### PR TITLE
chore(dump): add dump tests

### PR DESCRIPTION
**Pull request by irevoire**
Add tests on the import of dump v1, v2, v3 and v4.

Since the dumps are slow to decompress, I made the `flate2` crate always compile in optimized.
And since they're also slow to index, I also made the `milli` crate always compile in optimized. What do you think of this @MarinPostma?
Should we keep milli unoptimized in case it could help us debug some things? :eyes:

### CODE REVIEW

**Review by MarinPostma — DISMISSED** [review]
Good job!

**Comment by irevoire** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
bors merge

**Review by Kerollmops — COMMENTED** [review]
> And since they're also slow to index, I also made the `milli` crate always compile in optimized. What do you think of this @MarinPostma?

Even compile in optimized mode, the crate keeps the debug symbol. The only thing that can be an issue is the inlining of the functions but it shouldn't be a problem since errors are always returned and panics returns a good enough backtrace.

What's funny is that the CI doesn't pass due to the newly added dumps test.

**Comment by curquiza** [comment]
The first fail (on macos) is not dumps, but snapshots
The second one (on windows) is indeed due to dumps

**Review by MarinPostma — APPROVED** [review]
Gj 👍

**Comment by irevoire** [comment]
bors merge

**Comment by irevoire** [comment]
bors merge

**Comment by irevoire** [comment]
Hey @curquiza, actually the dump that made the v0.27 fail had nothing special. And funny thing, if this PR had been merged earlier, it would have caught that bug 😬 
As you can see, we can’t merge this PR because it fails on the same bug: `Meilisearch (v0.27.0) failed to infer the version of the database. Please consider using a dump to load your data.`
And thus, we need to backport #2396 to main to be able to merge this PR.

**Comment by curquiza** [comment]
> Hey @curquiza, actually the dump that made the v0.27 fail had nothing special. And funny thing, if this PR had been merged earlier, it would have caught that bug 😬

I'm happy and sad at the same time 😂 

> And thus, we need to backport https://github.com/meilisearch/meilisearch/pull/2396 to main to be able to merge this PR.

Its already done with https://github.com/meilisearch/meilisearch/pull/2404

Trying to run bors

bors try

**Comment by irevoire** [comment]
Finally rebased + fixed my tests, I think we can merge @MarinPostma 
bors try

File: meilisearch-http/Cargo.toml [hunk 1]
```diff
@@ -79,8 +79,10 @@ walkdir = "2.3.2"
 [dev-dependencies]
 actix-rt = "2.7.0"
 assert-json-diff = "2.0.1"
+manifest-dir-macros = "0.1.14"
 maplit = "1.0.2"
 paste = "1.0.6"
+reqwest = { version = "0.11.9", features = ["blocking", "rustls-tls"], default-features = false }
```
Comment (Kerollmops) [inline meilisearch-http/Cargo.toml:85]: Please remove this dependency.

File: meilisearch-http/Cargo.toml [hunk 2]
```diff
@@ -79,8 +79,10 @@ walkdir = "2.3.2"
 [dev-dependencies]
 actix-rt = "2.7.0"
 assert-json-diff = "2.0.1"
+manifest-dir-macros = "0.1.14"
```
Comment (Kerollmops) [inline meilisearch-http/Cargo.toml:82]: Do you always need this dep?

File: meilisearch-http/tests/dumps/data.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+use std::path::PathBuf;
+
+use manifest_dir_macros::exist_relative_path;
+
+pub fn get_v1_movies_raw() -> PathBuf {
```
Comment (Kerollmops) [inline meilisearch-http/tests/dumps/data.rs:5]: Could you please define an enum here and define only one function?

File: meilisearch-lib/src/index_controller/dump_actor/mod.rs [hunk 4]
```diff
@@ -256,13 +256,8 @@ fn extract_dump(
         .parent()
         .map(ToOwned::to_owned)
         .unwrap_or_else(|| ".".into());
-    if cfg!(windows) {
-        std::env::set_var("TMP", temp_path);
-    } else {
-        std::env::set_var("TMPDIR", temp_path);
-    }
 
-    let tmp_src = tempfile::tempdir()?;
```
Comment (Kerollmops) [inline meilisearch-lib/src/index_controller/dump_actor/mod.rs:265]: Do you really want to commit this?

**Review by Kerollmops — CHANGES_REQUESTED** [review]

File: meilisearch-lib/src/index_controller/dump_actor/mod.rs (hunk 4 above)
Comment (irevoire) [reply meilisearch-lib/src/index_controller/dump_actor/mod.rs:265]: yeos

File: meilisearch-http/Cargo.toml (hunk 2 above)
Comment (irevoire) [reply meilisearch-http/Cargo.toml:82]: yes

File: meilisearch-lib/src/index_controller/dump_actor/mod.rs (hunk 4 above)
Comment (Kerollmops) [reply meilisearch-lib/src/index_controller/dump_actor/mod.rs:265]: okeos

File: meilisearch-http/Cargo.toml (hunk 1 above)
Comment (irevoire) [reply meilisearch-http/Cargo.toml:85]: done

File: meilisearch-http/tests/dumps/data.rs (hunk 3 above)
Comment (irevoire) [reply meilisearch-http/tests/dumps/data.rs:5]: done

**Review by Kerollmops — APPROVED** [review]
That's much better, thank you very much for having taken the time to do this!

bors merge

**Pull request merged**